gloo/transport/ibverbs/buffer.cc

```diff
@@ -59,8 +59,28 @@ Buffer::Buffer(Pair* pair, int slot, void* ptr, size_t size)
 }
 
 Buffer::~Buffer() {
-  GLOO_ENFORCE_EQ(sendPending_, 0, "Destructing buffer expecting completions");
-  ibv_dereg_mr(mr_);
+  int sendPending = 0;
+  {
+    std::unique_lock<std::mutex> lock(m_);
+    sendPending = sendPending_;
+  }
+
+  if (sendPending > 0) {
+    GLOO_WARN(
+        "Destructing buffer with pending sends, sendPending_=", sendPending_);
+    try {
+      for (int i = 0; i < sendPending; i++) {
+        waitSend();
```

Comment:
Is this blocking so the destructor can hang?

Reply:
This actually isn't needed anymore -- there was an issue that's since been fixed. Changed this to just be a warning now